parquet-column/src/main/java/org/apache/parquet/column/ParquetProperties.java

```diff
@@ -97,21 +84,27 @@ public static WriterVersion fromString(String name) {
   private final int maxRowCountForPageSizeCheck;
   private final boolean estimateNextSizeCheck;
   private final ByteBufferAllocator allocator;
-
-  private final int initialSlabSize;
+  private final ValuesWriterFactory valuesWriterFactory;
 
   private ParquetProperties(WriterVersion writerVersion, int pageSize, int dictPageSize, boolean enableDict, int minRowCountForPageSizeCheck,
```

Comment:
Not sure I understand your question completely but if you take a look at my last commit: https://github.com/apache/parquet-mr/pull/342/commits/503958a90a0b2274f586a5fd229836149e7e2e23, you can see that there's a way to configure ValuesWriterFactories. To do so, you write your special ValuesWriterFactory (similar to what I've done in the unit tests) and make it extend ConfigurableFactory. When you do so, you have the Hadoop Config passed in which you can read & use. I did mull doing something like reading everything under `parquet.writer.writerProperties.*` but felt this was a cleaner approach.